### PR TITLE
Feat/Add limits to random module.

### PR DESCRIPTION
**Pull request by lhruby**
# Description
* Allow to specify limits for certain methods of the `random` module.
Note that `Bounds<T>` cannot be used here, as the module `types` depends on the module `random`, which would lead to cyclic dependencies.

## Type of change
- New feature (non-breaking change which adds functionality)

## Checklist before requesting a review
- [x] I have performed a self-review of my code.
- [x] If it is a core feature, I have added thorough tests.
- [x] If this is a new component I have added examples.
- [ ] I updated the README and related documentation.

### CODE REVIEW

File: modules/random/include/hephaestus/random/random_object_creator.h [hunk 1]
```diff
@@ -16,6 +16,7 @@
 
 #include "hephaestus/utils/concepts.h"
 #include "hephaestus/utils/exception.h"
+#include "hephaestus/types/bounds.h"
```
Comment (lhruby) [inline modules/random/include/hephaestus/random/random_object_creator.h:19]: @filippobrizzi there is an issue with our current module structure, which leads to cyclic dependencies. 

If we use `types::DummyType` to test other modules, then non of these modules must depend on types. If we add helper types like `Bounds<T>` to `types`, we cannot use them in most hephaestus modules. The random module is a good example.

Solutions: we either move DummyType into a separate module, like a test helper, or we try to formulate all helper types in terms of utility etc. The latter sounds more appealing.

`utility::???::Bounds<T>` <-- suggestions?

Comment (filippobrizzi) [reply modules/random/include/hephaestus/random/random_object_creator.h:19]: makes sense, actually `Bounds` is not really a type, it is a helper/utils I would say.

Comment (lhruby) [reply modules/random/include/hephaestus/random/random_object_creator.h:19]: Any suggestion for a folder/namespace for `Bounds`? utils::types, utils::data, utils::ranges, ...

Comment (filippobrizzi) [reply modules/random/include/hephaestus/random/random_object_creator.h:19]: I vote for `data`

**Review by filippobrizzi — APPROVED** [review]

**Pull request merged**